### PR TITLE
Replace some uses of the deprecated ReflectiveClassBuildItem constructors

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java [hunk 1]
```diff
@@ -120,18 +120,18 @@ void register(
...
             BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
         reflectiveClass.produce(
-                new ReflectiveClassBuildItem(true, false,
-                        "com.fasterxml.jackson.databind.ser.std.SqlDateSerializer",
+                ReflectiveClassBuildItem.builder("com.fasterxml.jackson.databind.ser.std.SqlDateSerializer",
                         "com.fasterxml.jackson.databind.ser.std.SqlTimeSerializer",
                         "com.fasterxml.jackson.databind.deser.std.DateDeserializers$SqlDateDeserializer",
                         "com.fasterxml.jackson.databind.deser.std.DateDeserializers$TimestampDeserializer",
-                        "com.fasterxml.jackson.annotation.SimpleObjectIdResolver"));
+                        "com.fasterxml.jackson.annotation.SimpleObjectIdResolver").methods(true).build());
```
Comment (gsmet) [inline extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java:127]: For next time, you can even use the `methods()` shortcut.

File: extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java [hunk 2]
```diff
@@ -221,7 +220,8 @@ void register(
             AnnotationValue value = resolverInstance.value("value");
             if (value != null) {
                 // Add the type-id-resolver class
-                reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, value.asClass().name().toString()));
+                reflectiveClass
+                        .produce(ReflectiveClassBuildItem.builder(value.asClass().name().toString()).methods(true).build());
```
Comment (gsmet) [inline extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java:224]: Shouldn't `fields()` be true too? Or it's not needed?

**Review by gsmet — COMMENTED** [review]
Added a small question.

File: extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java (hunk 1 above)
Comment (geoand) [reply extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java:127]: Yeah, I saw that, but honestly it feels weird to me :)

File: extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java (hunk 2 above)
Comment (geoand) [reply extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java]: Well spotted! Fixed

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Given things are failing in `native`, I wonder if the failures could be related.

**Comment by geoand** [comment]
I don't think the `main` integration test even depends on any of these, but I'll take a closer look tomorrow

**Comment by geoand** [comment]
The failures are definitely related...

**Comment by geoand** [comment]
It turned out that the first commit contained an unintentional change (that was meant for another PR) that was causing the failure

**Pull request merged**